content/technotes/2020-02-14-opentripplanner.Rmd

```diff
@@ -0,0 +1,100 @@
+
+The aim of the OpenTripPlanner for R package[^1] to make it easy to set up and use a multimodal trip planner in R. [OpenTripPlanner](https://www.opentripplanner.org/) (OTP) is an open-source multimodal trip planner written in Java. It uses OpenStreetMap for walking, cycling, and driving directions and uses [GTFS](https://developers.google.com/transit/gtfs) files for public timetables. The R package makes it easy to set up and use OpenTripPlanner on your local computer or connecter to a sever running OpenTripPlanner.
+
+## Prerequisites and Installation 
+
+A full list of the prerequisites and instructions is available in the  [vignettes](https://docs.ropensci.org/opentripplanner/articles/opentripplanner.html), but the key one is that OTP requires Java 8 specifically, not any earlier or later versions. Once you have Java installed,  install the package in the normal way.
+
+```{r installCRAN, eval=FALSE} 
+install.packages("opentripplanner") 
+library(opentripplanner) 
+``` 
+
+## Building your first trip planner
+
+The package comes with demo data for the Isle of Wight, which can be downloaded and built using a few simple commands. For a full explanation, including how to customize your setup and build OTP for other places, see the [vignettes](https://docs.ropensci.org/opentripplanner/articles/opentripplanner.html). 
+
+```{r, eval=FALSE} 
+path_data <- file.path(tempdir(), "OTP") # Make a folder to store the data 
+dir.create(path_data)  
+path_otp <- otp_dl_jar(path_data)        # Download the OTP 
+otp_dl_demo(path_data)                   # Download the demo data 
+log1 <- otp_build_graph(otp = path_otp, 
+                        dir = path_data) # Built the demo data into a graph 
+log2 <- otp_setup(otp = path_otp, 
+                  dir = path_data)       # Start OTP 
+``` 
+
+The whole process should only take a few minutes to run, and then the OTP web interface will launch in your browser. You can use the web interface just like any other journey planner, but to get the results into R you need to connect R to the OTP and then make your requests through R.
+
+```{r, eval = FALSE} 
+otpcon <- otp_connect()                 # Connect R to OTP 
+route <- otp_plan(otpcon,               # Route between two lon/lat coordinates 
+                  fromPlace = c(-1.17502, 50.64590),  
+                  toPlace = c(-1.15339, 50.72266),
+                  mode = c("WALK","TRANSIT")) 
+```
+
+If you have the `tmap` package installed, you can view the route within RStudio. Notice the use of `sf::st_zm(route)` this is because `tmap` does not currently support XYZ coordinates. 
```

Comment:
```suggestion
If you have the `tmap` package installed, you can view the route within RStudio. Notice the use of `sf::st_zm(route)`. This is because `tmap` does not currently support XYZ coordinates. 
```

Fixed grammar by making 2 sentences